river/tree/splitter/histogram_splitter.py

```diff
@@ -76,28 +76,29 @@ def best_evaluated_split_suggestion(
         thresholds = list(decimal_range(start=low, stop=high, num=n_thresholds))
         cdfs = {y: hist.iter_cdf(thresholds) for y, hist in self.hists.items()}
 
+        total_weight = sum(pre_split_dist.values())
         for at in thresholds:
 
             l_dist = {}
             r_dist = {}
-            total_weight = sum(pre_split_dist.values())
 
             for y in pre_split_dist:
-                p_xy = next(cdfs[y]) if y in cdfs else 0.0  # P(x < t | y)
-                p_y = pre_split_dist[y] / total_weight  # P(y)
-                l_dist[y] = total_weight * p_y * p_xy  # P(y | x < t)
-                r_dist[y] = total_weight * p_y * (1 - p_xy)  # P(y | x >= t)
-
-                post_split_dist = [l_dist, r_dist]
-                merit = criterion.merit_of_split(pre_split_dist, post_split_dist)
-
-                if best_suggestion is None or merit > best_suggestion.merit:
-                    num_att_binary_test = NumericAttributeBinaryTest(
-                        att_idx, at, equal_passes_test=False
-                    )
-                    best_suggestion = AttributeSplitSuggestion(
-                        num_att_binary_test, post_split_dist, merit
-                    )
+                if y in cdfs:
```

Comment:
Could you elaborate on this? Code after `if` is no longer part of the loop?

Reply:
This was a bug :/

We need to iterate over all classes in `pre_split_dist` before evaluating the resulting post-split distribution. We didn't have tests for the different splitters until now so that one escaped my attention.